mono/mini/exceptions-amd64.c

```diff
@@ -875,7 +877,8 @@ altstack_handle_and_restore (MonoContext *ctx, MonoObject *obj, guint32 flags)
 	gboolean nullref = (flags & 2) != 0;
 
 	if (!ji || (!stack_ovf && !nullref))
-		mono_handle_native_crash ("SIGSEGV", ctx, NULL);
+		if (mono_dump_start ())
```

Comment:
if `mono_dump_start()` is false here, we don't want to fall through (to managed exception unwinding), we want the process to die.

Reply:
I don't think handle_native_crash () can return, the current code kinda assumes it doesn't.

Reply:
We talked on slack and it looks like it can return if crash chaining is TRUE (xamarin-macios sets it to TRUE, for example).  So this needs to be:
```
       if (mono_dump_start ())
            mono_handle_native_crash ("SIGSEGV", ...);
      // if couldn't dump or if mono_handle_native_crash returns, abort
      abort ();
```